App.Stocks/Services/ValidateServices.cs

```diff
@@ -0,0 +1,47 @@
+﻿using App.Stocks.ModelsView;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+
+namespace App.Stocks.Services
+{
+	public class ValidateServices : IValidateServices
+	{
+		public void ValidateCompany(Company company, bool IsOpenStocks = true)
+		{
+			if (company == null)
+			{
+				throw new HttpListenerException(400, "Company doesn't exist");
+			}
+			if (!IsOpenStocks)
+			{
+				throw new HttpListenerException(400, "You can't see stocks this company");
```

Comment:
HttpListenerException -> Exception. Because business logic should know nothing about API part